### PR TITLE
Fix Media Element Windows CTD

### LINKED ISSUE #1920: [BUG] Windows Media Element CTD When using Multiple MediaElements
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

With two or more media elements on a page windows app will crash to desktop.

### Expected Behavior

It should playback video on demand and the app should not crash.

### Steps To Reproduce

1. Create a project with Media Element. 
2. Add a minimum of two items and launch it as a windows app. It will crash to desktop.

### Link to public reproduction project repository

https://github.com/ne0rrmatrix/MediaElementWindowsCTDBug

### Environment

```markdown
- .NET MAUI CommunityToolkit: 9.0.1
- OS:  Windows 10 Build 10.0.19041.0
- .NET MAUI: 8.0.6
```

### Anything else?

There is no crash report as it is not generated by crash. It just immediately crashes with no crash report. App just terminates.

### PR DESCRIPTION
**Pull request by ne0rrmatrix**
- Bug fix
 
 ### Description of Change ###
-Update font icon behavior to prevent CTD.

 ### Linked Issues ###

 - Fixes #1920

 ### PR Checklist ###

 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

Fixes crash where if you have multiple media elements on a page in windows the app will immediately crash to desktop. This was an issues with font icons and has been resolved with this PR.

### CODE REVIEW

**Comment by vhugogarcia** [comment]
Thanks @ne0rrmatrix for the PR. I believe changing the icon to a font icon makes sense. 

However, is Segoe Fluent Icons a system windows font? if yes, what will happen if that font does not exist in the OS? 🤔

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public MauiMediaElement(MediaPlayerElement mediaPlayerElement)
 
 		fullScreenButton = new Button
 		{
-			Content = fullScreenIcon,
+			Content = new FontIcon() { Glyph = "\uE740", FontFamily = new FontFamily("Segoe Fluent Icons") },
```
[the remark this answers is not in the document]
Comment (ne0rrmatrix) [reply src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:49]: We need to have it available or create another new `FontIcon` in full screen method. Either or will work.

**Comment by ne0rrmatrix** [comment]
> Thanks @ne0rrmatrix for the PR. I believe changing the icon to a font icon makes sense.
> 
> However, is Segoe Fluent Icons a system windows font? if yes, what will happen if that font does not exist in the OS? 🤔

This is a windows specific fix and the font icon is standard across current windows versions that we support.

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bijington) [reply src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:49]: Oh no worries, I couldn't see where else it was used but if it still is then I'll resolve this

**Comment by ne0rrmatrix** [comment]
Since we were having issues with a static readonly object I have removed it and it is now generated dynamically. Not sure if the second call would be impacted but I can't guess every use case. The only question now is do we want to that in case somehow we end up with two fullscreen media elements on the same page which would also generate the same bug? Not sure how that would happen but if it does it will crash.

If we do I can remove ```static readonly FontIcon exitFullScreenIcon = new() { Glyph = "\uE73F", FontFamily = new FontFamily("Segoe Fluent Icons") };``` and replace with dynamic calls too.

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs [hunk 2]
```diff
@@ -163,7 +161,7 @@ void OnFullScreenButtonClick(object sender, RoutedEventArgs e)
 				popup.Child = null;
 				fullScreenGrid.Children.Clear();
 			}
-			fullScreenButton.Content = exitFullScreenIcon;
+			fullScreenButton.Content = new FontIcon() { Glyph = "\uE740", FontFamily = new FontFamily("Segoe Fluent Icons") };
```
Comment (ne0rrmatrix) [inline src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:164]: Fix bug where in some scenarios app will crash on exit when not in full screen

**Comment by brminnick** [comment]
I too am a bit nervous hard coding a font here. The smells like a potential for a future bug.

@ne0rrmatrix - could you add the font as a resource to the Windows platform in `CommunityToolkit.Maui.MediaElement` (ie add the `.ttf` or `.otf` file as an embedded resource to ensure the font always ships with the NuGet package? This is probably overkill, but it'll prevent a future bug if Windows removes/changes Segoe Fluent Icons).

**Comment by ne0rrmatrix** [comment]
> Segoe Fluent Icons

Updated as requested. Font added as resource for windows.

**Comment by bijington** [comment]
> I too am a bit nervous hard coding a font here. The smells like a potential for a future bug.
> 
> 
> 
> @ne0rrmatrix - could you add the font as a resource to the Windows platform in `CommunityToolkit.Maui.MediaElement` (ie add the `.ttf` or `.otf` file as an embedded resource to ensure the font always ships with the NuGet package? This is probably overkill, but it'll prevent a future bug if Windows removes/changes Segoe Fluent Icons).

I've tried to look but have failed to find the licensing rules around the specific font. Are we allowed to distribute the font?

**Comment by ne0rrmatrix** [comment]
License can be found by downloading Mac download link here. It is in zip file. Scroll down page and look for mac assets. https://learn.microsoft.com/en-us/windows/apps/design/style/segoe-ui-symbol-font

**Comment by ne0rrmatrix** [comment]
I just double checked and google first result is that "Segoe UI" is a windows system font. The MDL2 icons I reference are system icons that are a core part of the OS. I'm not sure how we move forward if this is an issue. We can of course close this PR and go back and revert the previous one that added this use of fonts and look for an alternative method of fixing the previous issue? Can we not rely on system resources to just be available?

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs [hunk 3]
```diff
@@ -31,8 +31,9 @@ public class MauiMediaElement : Grid, IDisposable
 	readonly Grid buttonContainer;
 	readonly Button fullScreenButton;
 	readonly MediaPlayerElement mediaPlayerElement;
-	static readonly FontIcon fullScreenIcon = new() { Glyph = "\uE740", FontFamily = new FontFamily("Segoe Fluent Icons") };
-	static readonly FontIcon exitFullScreenIcon = new() { Glyph = "\uE73F", FontFamily = new FontFamily("Segoe Fluent Icons") };
+	// Cannot be static readonly because we need to be able to add icon to multiple instances of the button
+	FontIcon fullScreenIcon = new() { Glyph = "\uE740", FontFamily = new FontFamily("Segoe Fluent Icons") };
+	FontIcon exitFullScreenIcon = new() { Glyph = "\uE73F", FontFamily = new FontFamily("Segoe Fluent Icons") };
```
Comment (brminnick) [inline src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:36]: Can these still be `readonly`

```suggestion
	readonly FontIcon fullScreenIcon = new() { Glyph = "\uE740", FontFamily = new FontFamily("Segoe Fluent Icons") };
	readonly FontIcon exitFullScreenIcon = new() { Glyph = "\uE73F", FontFamily = new FontFamily("Segoe Fluent Icons") };
```

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs [hunk 4]
```diff
@@ -163,7 +164,7 @@ void OnFullScreenButtonClick(object sender, RoutedEventArgs e)
 				popup.Child = null;
 				fullScreenGrid.Children.Clear();
 			}
-			fullScreenButton.Content = exitFullScreenIcon;
+			fullScreenButton.Content = fullScreenIcon;
```
Comment (brminnick) [inline src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:167]: This was previously `exitFullScreenIcon`. Is changing it to `fullScreenIcon` a typo, or a bug fix?

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs [hunk 5]
```diff
@@ -182,7 +183,7 @@ void OnFullScreenButtonClick(object sender, RoutedEventArgs e)
 			mediaPlayerElement.Height = displayInfo.Height / displayInfo.Density;
 
 			Children.Clear();
-			fullScreenButton.Content = fullScreenIcon;
+			fullScreenButton.Content = exitFullScreenIcon;
```
Comment (brminnick) [inline src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:186]: This was previously `fullScreenIcon`. Is changing it to `exitFullScreenIcon` a typo, or a bug fix?

**Review by brminnick — COMMENTED** [review]
Thanks James! I owe you an apology - I see now that we were already using `Segoe Fluent Icons` in MediaElement and that this PR does _not_ introduce that change. (A lazy PR review on my part - sorry!)

I'm cool moving forward with this. I have a couple comments (below). After that, I'm cool approving/merginfg this 💯

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs (hunk 4 above)
Comment (ne0rrmatrix) [reply src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:167]: bug fix

File: src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs (hunk 5 above)
Comment (ne0rrmatrix) [reply src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs:186]: bug fix. They were swapped incorrectly in orginal PR.

**Review by brminnick — APPROVED** [review]
Thanks @ne0rrmatrix!!

**Review by brminnick — APPROVED** [review]

**Pull request merged**